File: Data Structures & Algorithms/is-anagram/submission-2.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char,int> m;

        for(int i=0;i<s.size();i++){
            m[s[i]]++;
        }

        for(int i=0;i<t.size();i++){
            if (m.find(t[i])==m.end()){
                return false;
            }else{
                m[t[i]]--;
                if (m[t[i]]==0){
                    m.erase(t[i]);
                }
            }
        }
        return m.size()==0;
    }
};

File: Data Structures & Algorithms/subsets/submission-2.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        sol=[]
        temp=[]
        def find(i:int,sol:[],temp:[],nums: List[int]) -> None:
            if (i==len(nums)):
                sol.append(temp.copy())
                return 
            
            find(i+1,sol,temp,nums)
            temp.append(nums[i])
            find(i+1,sol,temp,nums)
            temp.pop()
        
        find(0,sol,temp,nums)
        return sol


